cmd/katalyst-agent/app/agent/context.go

```diff
@@ -109,6 +111,31 @@ func (c *GenericContext) Run(ctx context.Context) {
 	<-ctx.Done()
 }
 
+// customizedPluginManager requires that handlers must be added before
+// it really runs to avoid logging error logs too frequently
+type customizedPluginManager struct {
+	enabled *atomic.Bool
+	pluginmanager.PluginManager
+}
+
+// Run successes iff at-least-one handler has been registered
+func (p *customizedPluginManager) Run(sourcesReady config.SourcesReady, stopCh <-chan struct{}) {
+	if p.enabled == nil || !p.enabled.Load() {
```

Comment:
should we deal with the condition that PluginManager.Run is called before PluginManager.AddHandler? 

It's similar to [shared_informer adds listener after it's started](https://github.com/kubernetes/client-go/blob/master/tools/cache/shared_informer.go#L611).

Reply:
in our case, the AddHandler must be called before Run, which is guaranteed by the agent framework.